src/components/ProductList/productContext.js

```diff
@@ -0,0 +1,34 @@
+import { createContext, useContext } from "react";
+import { productReducer } from "./productReducer";
+import { useReducer } from "react";
+export const ProductContext = createContext();
+
+export const ProductProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(productReducer, {
+    products: [],
+    sortBy: null,
+    showInventoryAll: true,
+    showFastDeliveryOnly: false
+  });
+
+  return (
+    <ProductContext.Provider
+      value={
+        (productReducer,
+        {
+          products: state.products,
+          sortBy: state.sortBy,
+          showInventoryAll: state.showInventoryAll,
+          showFastDeliveryOnly: state.showFastDeliveryOnly,

```

Comment:
No need to destructure the state here, you can do it in the components directly
```suggestion
          state,
```